### ISSUE TITLE
AfSigmoidBias activation function seems wrong

### ISSUE THREAD

**Issue by JoostHuizinga** [sferes2/nn2#4]
The activation function in AfSigmoidBias, defined in af.hpp, seems wrong. As you can see below, in the current implementation, only the bias (`trait<P>::single_value(this->_params`) is multiplied by lambda.

```
return 1.0 / (exp(-p + trait<P>::single_value(this->_params) * lambda) + 1);
```

However, I would assume that the sum of (-p + bias) should be multiplied by lambda. In code:

```
return 1.0 / (exp((-p + trait<P>::single_value(this->_params)) * lambda) + 1);
```

Also, since it is called a bias, shouldn't it be added to p before it becomes negative? In its current form, it seems that the bias would function more like a threshold.

**Comment by jbmouret** [comment]
Good catch! At first sight, you seem to be right (I am not sure we have
used this function a lot. Did you use it regularly?)

Le ven. 28 oct. 2016 01:32, Joost Huizinga <notifications@github.com> a
écrit :

**Comment by JoostHuizinga** [comment]
I don't think I have used this function in previous work; the retina experiment had the activation function defined within the experiment file itself, and the rest of my research used completely different types of activation functions.
